### DISCUSSION TITLE
Langfuse Trace with OpenAI Assistants

### DISCUSSION THREAD

**Discussion by marco-introvigne** [langfuse/langfuse#1513] [Ideas]
### Describe the feature or potential improvement

Currently langfuse is not working with the function of langchain `OpenAIAssistantRunnable`. When i am creating a new agent is not tracking the informations of the Cost and the Input/Output tokens. Would be a great feature to add

**Comment by marcklingen** [comment]
Hi @marco-introvigne, agree, full support of OpenAI assistants would be cool, let's track this idea here and see who else is interested

**Comment by juan-abia** [comment]
This would be awesome! I'm not super familiar with openai assistant API and I guess it would be hard to integrate with current langfuse cost tracking. But it'd be great if we could track tool usage cost.

For example, using openAI code interpreter costs a few cents per session, it'd be cool if we could track that cost in the trace

**Comment by clemra** [comment]
Hi all, we've written up a cookbook that outlines how you can use our [@observe decorator](https://langfuse.com/docs/sdk/python/decorators) in using Langfuse with the Assistants API: 

https://langfuse.com/guides/cookbook/integration_openai_assistants

**Comment by gablabelle** [comment]
Hello @marcklingen @clemra ! Brand new to Langfuse.

I'm going through the docs and it seems there is no JS/TS examples with Langfuse and Open AI Assistants API.

https://langfuse.com/faq/all/openai-assistant-api#jsts

> We do not yet have a published example on how to do this via the Langfuse JS/TS SDK. However, you can have a look at the Python example to understand the general approach: (1) create an assistant, (2) run it on a thread, and (3) observe the execution with Langfuse tracing.

Is this on your roadmap?

Could someone point me to which utility function(s) in the docs (https://js.reference.langfuse.com/) allow you to "observe" the execution (point #3) or to a simple idiomatic example about how we observe executions using JavaScript or TypeScript.

Many thanks

**Reply by marcklingen** [reply]
Yes, you'll need to use the low level sdk, have a look at the guide here: https://langfuse.com/docs/sdk/typescript/guide

**Comment by heresandyboy** [comment]
HI, I am hoping to get set up with langfuse to trace and cost our Open AI Assistants API usage.
I have been through the discord, langfuse docs and over the internet hoping to find anyone that has shared a successful setup in JS/TS.
Can anybody watching this issue let me know if they have come up with a working example, before I have to deep dive and try to figure it out from scratch?

**Reply by marcklingen** [reply]
Currently, there is no example available for JS/TS. Have you checked out the Python example as a reference?
